src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -96,10 +99,13 @@ public override void Contextualize(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(expression));
             }
 
-            var modelExplorer = ExpressionMetadataProvider.FromLambdaExpression(expression, ViewData, MetadataProvider);
-
-            return GenerateDropDown(modelExplorer, ExpressionHelper.GetExpressionText(expression), selectList,
```

Comment:
I'm trusting you that you're doing the right thing here. I feel like you're the only person that really knows what that is...